### PR TITLE
ArgumentKey impl for Testing API

### PR DESCRIPTION
**Pull request by szalpal**

### CODE REVIEW

File: dali/test/argument_key.h [hunk 1]
```diff
@@ -15,27 +15,34 @@
...
 
-class ArgumentKey : public std::pair<std::string, std::string> {
+/**
+ * Class, that uniquely identifies Argument in graph
+ * (e.g. graph of Operators)
+ */
+class ArgumentKey {
  public:
-  using Base = std::pair<std::string, std::string>;
+  ArgumentKey(const char *arg_name) noexcept;  // NOLINT (non-explicit ctor)
```
[the remark this answers is not in the document]
Comment (mzient) [reply dali/test/argument_key.h]: So we can promote a string literal directly to argument key in a brace-enclosed initializer, e.g:
`{ "key", value }`

File: dali/test/argument_key.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  return node_name_;
+}
+
+
+std::string ArgumentKey::arg_name() const noexcept {
+  return arg_name_;
+}
+
+
+bool ArgumentKey::operator<(const ArgumentKey &rhs) const noexcept {
```
Comment (mzient) [inline dali/test/argument_key.cc]: Please make the one-liners inline.

Comment (szalpal) [reply dali/test/argument_key.cc]: Done

File: dali/test/argument_key.h [hunk 3]
```diff
@@ -15,27 +15,50 @@
...
 
 namespace dali {
 namespace testing {
 
-class ArgumentKey : public std::pair<std::string, std::string> {
+class ArgumentKey {
  public:
-  using Base = std::pair<std::string, std::string>;
+
+  ArgumentKey(const char *arg_name) noexcept : // NOLINT (non-explicit ctor)
```
Comment (mzient) [inline dali/test/argument_key.h]: Style: when initializer lists is moved to a new line, move the colon, too.

File: dali/test/argument_key.h [hunk 4]
```diff
@@ -15,27 +15,50 @@
...
+          node_name_(), arg_name_(arg_name) {
+    assert(!arg_name_.empty());  // Arg name has been set either as an empty string or not set at all
+  }
+
+
+  ArgumentKey(std::string node_name, std::string arg_name) noexcept :
```
Comment (mzient) [inline dali/test/argument_key.h]: Same as above - colon in new line.

File: dali/test/argument_key_test.cc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  EXPECT_EQ("", argk.node_name());
+
+  ArgumentKey opargk("op", "arg");
+  EXPECT_EQ("arg", opargk.arg_name());
+  EXPECT_EQ("op", opargk.node_name());
+}
+
+
+TEST(ArgumentKeyTest, ArgumentOperatorLessTest) {
+  ArgumentKey a1("arg1");
```
Comment (mzient) [inline dali/test/argument_key_test.cc]: Add test(s) with compound keys, preferably with first and second part equal:
`ArgumentKey a1("A", "X"), a2("A", "Y"), a3("B", "X");`
then compare `a1 < a2` and `a1 < a3'.

File: dali/test/argument_key.h (hunk 3 above)
Comment (szalpal) [reply dali/test/argument_key.h]: Done

File: dali/test/argument_key.h (hunk 4 above)
Comment (szalpal) [reply dali/test/argument_key.h]: Done

File: dali/test/argument_key_test.cc (hunk 5 above)
Comment (szalpal) [reply dali/test/argument_key_test.cc]: Done

File: dali/test/argument_key.h [hunk 6]
```diff
@@ -15,27 +15,50 @@
 #ifndef DALI_TEST_ARGUMENT_KEY_H_
 #define DALI_TEST_ARGUMENT_KEY_H_
 
-#include <utility>
+#include <assert.h>
```
Comment (mzient) [inline dali/test/argument_key.h]: Pedantic ;) `#include <cassert>` Fix only if you change anything else along the way, otherwise don't bother.

**Pull request merged**